ros_ws/src/interpreter_pkg/interpreter_pkg/interpreter.py

```diff
@@ -0,0 +1,168 @@
+import os
+import rclpy
+from rclpy.node import Node
+from example_interfaces.msg import String
+
+import langchain
+from langchain.chains import LLMChain
+from langchain.chat_models import ChatOpenAI
+from langchain.llms import OpenAI
+from langchain.prompts import (ChatPromptTemplate, HumanMessagePromptTemplate,
+                                SystemMessagePromptTemplate)
+from langchain.schema import  HumanMessage, SystemMessage
+
+from ament_index_python.packages import get_package_share_directory
+
+data_dir = os.path.join(get_package_share_directory("interpreter_pkg"))
+OpenAI.api_key=os.environ["OPENAI_API_KEY"]
+MODEL_TO_USE="gpt-4"
+
+class LLM_Object:
+    def __init__(self, context_file: str, model:str, temperature:int):
+        with open(os.path.join(data_dir,context_file), 'r') as file:
+            context= file.read().replace('\n', '')
+        self.context=context
+        print("context is: "+self.context)
+        self.rolling_context=self.context
+        self.temp=temperature
+        self.chat=ChatOpenAI(model_name=model)
+        self.system_message=SystemMessage(content=self.context)
+
+    def reset_context(self):
+        self.rolling_context=[self.system_message]
+        
+class TerpNode(Node):
+    def __init__(self):
+        super().__init__('interpreter')
+        self.get_logger().info("Terp Started")
+        #define elements
+        self.refiner=LLM_Object("refiner_context.prompt",MODEL_TO_USE,0) 
+        self.judger=LLM_Object("judge_context.prompt",MODEL_TO_USE, 0) 
+        self.extractor=LLM_Object("extractor_context.prompt",MODEL_TO_USE,0) 
+        self.new_request=True
+        self.attempts_to_understand=0
+        self.target=""
+        self.rolling_context=[]
+        self.last_suggestion=""
+        # define publishing
+        self.target_publisher_=self.create_publisher(String,'target',10)
+        self.chat_publisher_=self.create_publisher(String, 'clarifying_question',10)
+        #define subscriptions
+        self.subscriber_=self.create_subscription(String, 'user_request',self.get_request,10)
+
+    def get_request(self, request_msg: String): #callback for subscribing to user response
```

Comment:
Typically function documentation is underneath using `"""` block comments. Doing it this way lets hover over and other tools present other programmers with the documentation in easily readable format immediately. Like this:

```
def my_function():
   """
   my_function accepts the stuff and does the thing.
   """
```

Reply:
fixed